app/services/container_topology_service.rb

```diff
@@ -47,8 +47,36 @@ def build_topology
     topology
   end
 
-  def build_entity(id, name, kind)
-    {:metadata => {:id => id, :name => name}, :kind => kind}
+  def build_entity_data(entity, kind)
+    status = entity_status(entity, kind)
+
+    case kind
+    when 'VM', 'Host'
+      id = entity.uid_ems
+    else
+      id = entity.ems_ref
+    end
+
+    {:metadata => {:id => id}, :name => entity.name, :status => status, :kind => kind}
+  end
+
+  def entity_status(entity, kind)
+    case kind
```

Comment:
the existence of `status` local var is not needed here

think about all of the function being en expression, please